src/main/scala/zio/zmx/diagnostics/ZMXClient.scala

```diff
@@ -17,63 +17,70 @@
 package zio.zmx.diagnostics
 
 import java.nio.charset.StandardCharsets
-
+import java.nio.charset.StandardCharsets._
 import zio.console._
-import zio.nio.channels.SocketChannel
-import zio.nio.core.{ Buffer, ByteBuffer, SocketAddress }
-import zio.zmx.diagnostics.parser.Resp
-import zio.{ Chunk, Task, ZIO }
+import zio.{ Chunk, Task, ZIO, ZManaged }
+
+import java.net.InetSocketAddress
+import java.nio.ByteBuffer
+import java.nio.channels.SocketChannel
+import scala.annotation.tailrec
 
 object ZMXClient {
 
   /**
    * Generate message to send to server
    */
-  def generateRespCommand(args: Chunk[String]): Chunk[Byte] =
-    Resp.Array(args.map(Resp.BulkString)).serialize
+  def generateRespCommand(args: List[String]): String = {
+    val protocol = new StringBuilder().append("*").append(args.length).append("\r\n")
+
+    args.foreach { arg =>
+      val length = arg.getBytes(UTF_8).length
+      protocol.append("$").append(length).append("\r\n").append(arg).append("\r\n")
+    }
+
+    protocol.result
+  }
 
 }
 
 class ZMXClient(config: ZMXConfig) {
 
-  def sendCommand(args: Chunk[String]): ZIO[Console, Exception, String] =
-    nioRequestResponse(ZMXClient.generateRespCommand(args)).refineToOrDie[Exception]
+  def sendCommand(args: List[String]): ZIO[Console, Exception, String] = {
+    val sending: String = ZMXClient.generateRespCommand(args)
+    nioRequestResponse(sending).refineToOrDie[Exception]
+  }
 
-  private def nioRequestResponse(req: Chunk[Byte]): Task[String] = {
-    def drainBuffer(acc: Chunk[String], buffer: ByteBuffer): Task[Chunk[String]] =
-      for {
-        hasRemaining <- buffer.hasRemaining
-        result       <- if (hasRemaining)
-                          buffer
-                            .withJavaBuffer(buf => ZIO.succeed(acc :+ StandardCharsets.UTF_8.decode(buf).toString))
-                            .flatMap(drainBuffer(_, buffer))
-                        else ZIO.succeed(acc)
-      } yield result
+  private def nioRequestResponse(req: String): Task[String] = {
+    @tailrec
+    def drainBuffer$(acc: Chunk[String], buffer: ByteBuffer): Chunk[String] =
+      if (buffer.hasRemaining) {
+        drainBuffer$(acc :+ StandardCharsets.UTF_8.decode(buffer).toString, buffer)
+      } else acc
```

Comment:
I think we should keep operating on bytes until the very end.
Here, I guess you can end up with malformed Unicode if there was non-ASCII encoded character and boundary of two buffers ended up splitting those code points.

Reply:
Ok, 👌, not sure it's in the scope of the issue, I wanted to stay close to the original code and remove the dependency.

Reply:
Sure, it can be revisited later. Just thinking out loud 😄